pkg/spec/spec.go

```diff
@@ -31,17 +32,18 @@ type Function struct {
 
 // FunctionSpec contains func specification
 type FunctionSpec struct {
-	Handler             string             `json:"handler"`               // Function handler: "file.function"
-	Function            string             `json:"function"`              // Function file content or URL of the function
-	FunctionContentType string             `json:"function-content-type"` // Function file content type (plain text, base64 or zip)
-	Checksum            string             `json:"checksum"`              // Checksum of the file
-	Runtime             string             `json:"runtime"`               // Function runtime to use
-	Type                string             `json:"type"`                  // Function trigger type
-	Topic               string             `json:"topic"`                 // Function topic trigger (for PubSub type)
-	Schedule            string             `json:"schedule"`              // Function scheduled time (for Schedule type)
-	Timeout             string             `json:"timeout"`               // Maximum timeout for the function to complete its execution
-	Deps                string             `json:"deps"`                  // Function dependencies
-	Template            v1.PodTemplateSpec `json:"template" protobuf:"bytes,3,opt,name=template"`
+	Handler                 string                           `json:"handler"`               // Function handler: "file.function"
+	Function                string                           `json:"function"`              // Function file content or URL of the function
+	FunctionContentType     string                           `json:"function-content-type"` // Function file content type (plain text, base64 or zip)
+	Checksum                string                           `json:"checksum"`              // Checksum of the file
+	Runtime                 string                           `json:"runtime"`               // Function runtime to use
+	Type                    string                           `json:"type"`                  // Function trigger type
+	Topic                   string                           `json:"topic"`                 // Function topic trigger (for PubSub type)
+	Schedule                string                           `json:"schedule"`              // Function scheduled time (for Schedule type)
+	Timeout                 string                           `json:"timeout"`               // Maximum timeout for the function to complete its execution
+	Deps                    string                           `json:"deps"`                  // Function dependencies
+	Template                v1.PodTemplateSpec               `json:"template" protobuf:"bytes,3,opt,name=template"`
+	HorizontalPodAutoscaler v2alpha1.HorizontalPodAutoscaler `json:"horizontalPodAutoscaler" protobuf:"bytes,3,opt,name=horizontalPodAutoscaler"`
```

Comment:
just a suggestion: we can name it a bit more abstract:  "Autoscaling Rule" or something like that. When using the CLI, it's okay because it's transparent. However if user wants to deploy function via the CRD object, HPA would probably not a good name (too specific)

Reply:
mm, but in case a user wants to specify it via the CRD object, the clearer it is the better, isn't it?

Reply:
I dont' have strong opinion on that. Maybe we leave it there for users feedback.